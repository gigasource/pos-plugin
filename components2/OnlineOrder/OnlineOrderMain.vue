<script>
import { onBeforeMount } from 'vue'
import { renderPendingOrdersFactory } from "./online-order-main-pending-orders-render";
import { renderKitchenOrdersFactory } from "./online-order-main-kitchen-orders-render";
import {
  declineOrder,
  completeOrder,
  loadOrders
} from "./online-order-main-logic-be";
import {
  genScopeId
} from "../utils";

export default {
  setup(props, { emit }) {
    onBeforeMount(async () => {
      await loadOrders()
    })

    const {
      renderPendingOrders
    } = renderPendingOrdersFactory()
    const {
      renderKitchenOdrers
    } = renderKitchenOrdersFactory()

    return genScopeId(() => (
      <div class="main">
        {renderPendingOrders()}
        {renderKitchenOdrers()}
        <dialog-complete-order onCompleteorder={completeOrder} onDeclineorder={declineOrder}></dialog-complete-order>
        <dialog-text-filter label="Reason"></dialog-text-filter>
      </div>
    ))
  }
}
</script>

<style scoped lang="scss">
.main {
  background-image: url('/plugins/pos-plugin/assets/out.png');
  width: 100%;
  height: 100%;
  display: flex;
  padding: 16px;

  .g-card {
    border-radius: 4px;
    border: solid #9e9e9e 0.05em;
    margin-bottom: 6px;
    font-size: 14px;

    .g-btn-bs {
      border-radius: 2px;
    }

    .g-card-title {
      padding-bottom: 8px;
    }

    .g-card-text {
      padding: 0 16px 16px;
    }

    .g-card-actions {
      padding: 0 8px 16px;

      .g-btn-bs {
        text-transform: capitalize;
      }
    }
  }
}

.header {
  display: flex;
  flex-wrap: wrap;
  font-size: 15px;
  font-weight: 700;
  margin-bottom: 12px;
}

.content {
  height: calc(100% - 34px);
  overflow: hidden scroll;

  &::-webkit-scrollbar {
    display: none;
  }
}

.pending-orders {
  width: 40%;
  height: 100%;
  overflow: hidden;

  &--empty {
    height: 50%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: white;
    border: 0.5px solid #9E9E9E;
    border-radius: 4px;

    p {
      font-size: 12px;
      color: #9E9E9E;
      margin-top: 8px;
    }
  }

  &--forward-store {
    display: flex;
    justify-content: center !important;
    align-items: center !important;
    padding: 2px !important;
    margin: -8px 16px 12px;
    text-align: center;
    background-color: #E1F5FE;
    border-radius: 12px !important;
  }

  &--title {
    align-items: flex-start !important;
    flex-wrap: nowrap !important;
  }

  &--call {
    width: 100%;
    background-color: white;
    border: 1px solid #9e9e9e;
    border-radius: 4px;
    padding: 16px;
    margin-bottom: 8px;

    &-title {
      display: flex;
      align-items: center;
      font-size: 14px;
      font-weight: bold;
    }

    &-buttons {
      display: flex;
      align-items: center;
      margin-top: 4px;

      .g-btn-bs {
        margin: 0;
        padding: 18px;
      }
    }
  }
}

.kitchen-orders {
  width: 60%;

  &--empty {
    height: 30%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: white;
    border: 0.5px solid #9E9E9E;
    border-radius: 4px;

    p {
      font-size: 12px;
      color: #9E9E9E;
      margin-top: 8px;
    }
  }

  &__timer {
    background: #EFEFEF;
    border: 1px solid #C4C4C4;
    border-radius: 2px;
    padding: 0 8px;
  }
}

.btn-clicked {
  background: #FFFFFF;
  border: 1px solid #536DFE !important;
  border-radius: 2px;
  color: #536DFE !important;
}

.options {
  background: #FFFFFF;
  border: 1px solid #D3D3D3;
  border-radius: 2px;

  .option {
    background-color: #FFF;
    font-size: 14px;
    line-height: 20px;
    padding: 12px 16px;
    text-align: left;
    user-select: none;
    cursor: pointer;
    color: #201F2B;

    &:hover {
      background-color: #EFEFEF;
    }
  }
}

.progress-remaining {
  color: #E57373;
  position: absolute;
  top: 3px;
  right: 3px;
  font-size: 15px;
  width: 32px;
  text-align: center;
}

@media screen and (max-width: 1023px) {
  .main {
    .g-card {
      .g-card-title {
        padding: 8px;
      }

      .g-card-text, .g-card-actions {
        padding: 0 8px 8px;

        & > .g-btn-bs {
          margin: 0 4px;
        }
      }
    }

    .pending-orders--empty > p {
      text-align: center;
    }

    .pending-orders--title {
      flex-wrap: wrap !important;
    }

    .pending-orders--btn-price {
      flex-direction: column;
      align-items: center;

      & > img {
        margin-right: 0 !important;
      }
    }

    .pending-orders--call-title {
      line-height: 1.2;

      div span {
        display: none;
      }
    }

    .pending-orders--call-buttons .g-btn-bs {
      padding: 4px;
    }
  }

}
</style>
