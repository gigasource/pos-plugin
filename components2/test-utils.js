import "../jest.setup";
import _ from 'lodash';
import {config, mount} from "@vue/test-utils";
import {hooks} from "./OrderView/pos-logic";
import {disablePay, payBtnClickable, payPrintMode, showIcon} from "./OrderView/pos-logic-be";
import StubFactory from './StubFactory';

jest.mock("vue-i18n", () => {
  return {
    useI18n() {
      return {
        t(t) {
          return t; // [Consider] return `t('${t}')` to indicate that i18n has been used
        },
        locale: "en"
      };
    }
  };
});
jest.mock("vue-router", () => {
  return {
    useRoute() {
      return {
        query: {
          type: 'default'
        }
      };
    },
    useRouter() {
      return {
        push() {
        }
      };
    }
  };
});

const dayjs = require("dayjs");
export const orm = require("schemahandler");
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const {stringify} = require("schemahandler/utils");
//jest.useFakeTimers("modern").setSystemTime(new Date("2021-01-01").getTime());
require("mockdate").set(new Date("2021-01-01").getTime());

const delay = require("delay");

export const foodTax = {taxes: [5, 10]};
export const drinkTax = {taxes: [16, 32]};

export const cola = {name: "Cola", price: 1.3, quantity: 1, ...drinkTax};
export const fanta = {name: "Fanta", price: 2, quantity: 1, ...drinkTax};
export const rice = {name: "Rice", price: 10, quantity: 1, ...foodTax};
export const ketchup = {name: "Add Ketchup", price: 3, quantity: 1};

beforeAll(async () => {
  orm.connect({uri: "mongodb://localhost:27017"}, "myproject");
  orm.registerSchema("Order", {
    inProgress: Boolean,
    items: [{}],
    table: Number
  });
  await orm("Order").deleteMany();
  await orm("Commit").deleteMany();

  orm.plugin(syncPlugin);
  orm.plugin(syncFlow);
  orm.registerCommitBaseCollection("Order");
  orm.plugin(require("../backend/commit/orderCommit"));
  orm.registerCollectionOptions("Order");
  orm.emit("commit:flow:setMaster", true);
});

afterEach(async () => {
  await orm("Order")
    .deleteMany()
    .direct();
  await orm("Commit")
    .deleteMany()
    .direct();
});

export let actions = [];
["showOrderReceipt", "pay", "printOrder"].forEach(e =>
  hooks.on(e, () => actions.push(e))
);
export let expectArray = () => [
  "payBtnClickable",
  payBtnClickable.value,
  "payPrintMode",
  payPrintMode.value,
  "showIcon",
  showIcon.value,
  "disablePay",
  disablePay.value,
  actions
];
//</editor-fold>

//<editor-fold desc="component">
export let component, wrapper;

export function setComponent(_component) {
  component = _component;
}

/**
 *
 * @param _component
 * @param options
 * @param useDefaults indicate whether _.defaults will be use for options
 * @returns {*}
 * example: makeWrapper(Order2, {shallow: false});
 */
export const makeWrapper = (_component, options, useDefaults = false) => {
  const defaultOption = {
    props: {},
    shallow: true,
    global: {
      directives: {
        touch: true
      },
      stubs: {
        'choose-table-dialog': true,
        'dialog-change-value': true,
        'dialog-choose-popup-modifier': true,
        "dialog-config-order-item": true,
        'dialog-form-input': true,
        'dialog-multi-payment': true,
        'dialog-multi-payment2': true,
        'dialog-product-search-result': true,
        'dialog-text-filter': true,
        'g-avatar': true,
        'g-autocomplete': true,
        'g-badge': true,
        'g-btn': true,
        "g-btn-bs": true,
        'g-card': true,
        'g-card-title': true,
        'g-combobox': true,
        'g-dialog': true,
        'g-divider': true,
        "g-icon": true,
        'g-item-group': true,
        'g-menu': true,
        "g-overlay": true,
        'g-radio': true,
        'g-simple-table': true,
        'g-spacer': true,
        'g-switch': true,
        'g-table': true,
        'g-tabs': true,
        'g-tab-item': true,
        'g-text-field': true,
        'g-text-field-bs': true,
        'g-toolbar': true,
        'g-snackbar': true,
        'pos-order': true,
        'pos-quick-order-toolbar': true,
        //'pos-order-split-order': true,
        'pos-keyboard-full': true,
        'pos-order-keyboard': true,
        'pos-order-move-items': true,
        'pos-order-receipt': true,
        'pos-order-voucher-dialog': true,
        'pos-textfield-new': true,
        'g-grid-select': StubFactory('g-grid-select'),
        'pos-payment-screen-payment-methods2': true,
        'pos-payment-screen-keyboard2': true,
        'pos-restaurant-payment-order-detail2': true,
        'pos-restaurant-payment-toolbar2': true,
      },
      mocks: {
        t: a => a,
        $t: a => a,
        $filters: {
          formatCurrency(val, decimals = 2) {
            if (!val || isNaN(val) || Math.floor(val) === val) return val;
            return val.toFixed(decimals);
          }
        }
      }
    }
  }
  const optionResult = useDefaults ? _.defaults(options, defaultOption) : _.merge(defaultOption, options)
  wrapper = mount(component || _component, optionResult);
};

beforeAll(() => (config.renderStubDefaultSlot = true));
