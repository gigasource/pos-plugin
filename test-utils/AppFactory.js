export default function (component) {
	return {
		name: 'App',
		components: {
			Component: component
		},
		render() {
			return <Component/>
		}
	}
}
