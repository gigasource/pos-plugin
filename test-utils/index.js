const cmsFactory = require('./mockCms')
const syncFactory = require('./mockSync')

module.exports = {
	cmsFactory,
	syncFactory
}
