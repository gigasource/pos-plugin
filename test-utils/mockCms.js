const { Socket, Io } = require('schemahandler/io/io')
const orm = require('schemahandler/orm')
const Hooks = require('schemahandler/hooks/hooks')
const path = require('path')
const fs = require('fs')
const _ = require('lodash')

function cmsFactory(testName) {
  orm.connect({uri: "mongodb://localhost:27017"}, testName);
  const socketToFrontend = new Io()
  socketToFrontend.listen('frontend')
  const cms = {
    init: async function () {
      await this.initDemoData()
    },
    initDemoData: async function () {
      const dataPath = path.resolve(__dirname, './dataMock/demoData.json')
      const data = JSON.parse(fs.readFileSync(dataPath, 'utf-8'));

      for (const collection in data) {
        await cms.getModel(collection).remove({})
        if (data.hasOwnProperty(collection)) {
          await Promise.all(data[collection].map(async document => {
            await cms.getModel(collection).create(document)
          }))
        }
      }
    },
    orm,
    socket: socketToFrontend,
    getModel: function (modelName) {
      return orm(modelName)
    }
  }
  _.extend(cms, new Hooks())
  return cms
}

module.exports = cmsFactory
