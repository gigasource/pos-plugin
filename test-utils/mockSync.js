const cmsFactory = require('./mockCms')
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const syncTransporter = require('schemahandler/sync/sync-transporter');
const { Socket, Io } = require('schemahandler/io/io')

async function syncFactory(testName, numberOfNode) {
	const cms = []
	for (let i = 0; i < numberOfNode; i++) {
		const _cms = cmsFactory(`${testName}${i}`)
		await _cms.init()
		_cms.orm.plugin(syncPlugin)
		_cms.orm.plugin(syncFlow)
		_cms.orm.plugin(syncTransporter)
		cms.push(_cms)
	}

	const masterIo = new Io()
	masterIo.listen('master')
	masterIo.on('connect', socket => {
		cms[0].orm.emit('initSyncForMaster', socket)
	})
	cms[0].orm.emit('commit:flow:setMaster', true)
	for (let i = 1; i < numberOfNode; i++) {
		const socket = new Socket()
		socket.connect('master')
		cms[i].orm.emit('initSyncForClient', socket)
	}
	return cms
}

module.exports = syncFactory
