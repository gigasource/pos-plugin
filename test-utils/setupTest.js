const { VueLoaderPlugin } = require('vue-loader')
const { WebpackManifestPlugin } = require('webpack-manifest-plugin');
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const path = require('path')
const webpack = require('webpack')

module.exports = async function (options) {
	const config = {
		entry: options.componentPath,
		output: {
			libraryTarget: 'commonjs2',
			publicPath: '', // prevent prefix 'auto'
			path: path.resolve(__dirname, '../../dist_test')
		},
		resolve: {
			extensions: ['.vue', '.jsx', '.js', '.json'],
			alias: {
				vue: "@vue/runtime-dom",
			},
		},
		externals: {
			vue: 'vue'
		},
		optimization: {
			minimize: false,
		},
		module: {
			rules: [
				{
					test: /\.vue$/,
					use: "vue-loader",
				},
				{
					test: /\.png$/,
					use: {
						loader: "url-loader",
						options: { limit: 8192 },
					},
				},
				{
					test: /\.js?$/,
					loader: "babel-loader"
				},
				{
					test: /\.css$/,
					use: [
						{
							loader: MiniCssExtractPlugin.loader,
						},
						"css-loader",
					],
				},
			],
		},
		plugins: [
			new MiniCssExtractPlugin({
				filename: "[name].css",
			}),
			new VueLoaderPlugin(),
			new WebpackManifestPlugin({ fileName: 'ssr-manifest.json' })
		]
	}

	return new Promise((resolve, reject) => {
		webpack(config, (err, stats) => {
			if (err) {
				return reject(err)
			}
			if (stats.hasErrors()) {
				return reject(stats.compilation.errors)
			}
			console.log('Build complete.')
			resolve()
		})
	})
}
