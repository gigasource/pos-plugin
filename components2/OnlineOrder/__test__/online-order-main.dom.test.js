import { wrapper, makeWrapper } from "../../test-utils";
import OnlineOrderMain from "../OnlineOrderMain";

describe("test call system ui", () => {
  it("should render", () => {
    makeWrapper(OnlineOrderMain);
    expect(wrapper.html()).toMatchInlineSnapshot(`
      <div class="main">
        <div class="pending-orders pr-2">
          <div class="header">
            <span>
              onlineOrder.pendingOrders
            </span>
            0
            <g-spacer-stub>
            </g-spacer-stub>
          </div>
          <div class="content">
          </div>
        </div>
        <div class="kitchen-orders pl-2">
          <div class="header">
            onlineOrder.sentToKitchen
          </div>
          <div class="content">
            <div class="kitchen-orders--empty">
              <p>
                onlineOrder.noKitchen
              </p>
            </div>
          </div>
        </div>
        <dialog-complete-order-stub modelvalue="[object Object]">
        </dialog-complete-order-stub>
        <dialog-text-filter-stub label="Reason">
        </dialog-text-filter-stub>
        <new-reservation-dialog-stub>
        </new-reservation-dialog-stub>
      </div>
    `);
  });
});
