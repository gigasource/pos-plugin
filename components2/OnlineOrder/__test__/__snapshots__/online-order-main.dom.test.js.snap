// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render online order should render 1`] = `
<div class="main">
  <div class="pending-orders pr-2">
    <div class="header">
      <span>
        onlineOrder.pendingOrders
      </span>
      <g-badge-stub class="ml-1"
                    inline="true"
                    modelvalue="true"
                    color="#4CAF50"
      >
      </g-badge-stub>
      <g-spacer-stub>
      </g-spacer-stub>
      <span style="color: rgb(211, 47, 47);">
        Modem not connected
      </span>
    </div>
    <div class="content">
      <g-card-stub elevation="0">
        <g-card-title-stub class="pending-orders--title">
          <div class="row-flex align-items-center flex-grow-1">
            <g-icon-stub>
              icon-take-away
            </g-icon-stub>
            <div class="fs-small-2 ml-1"
                 style="max-width: calc(100% - 24px); line-height: 1.2;"
            >
              <span class="fs-small fw-700 text-indigo-accent-2">
                #
              </span>
              ádf -123
            </div>
          </div>
          <div class="row-flex justify-end align-items-center r"
               style="flex: 0 0 auto;"
          >
            <span class="fw-700 fs-small ml-2 mr-2"
                  style="text-transform: uppercase;"
            >
              asap
            </span>
            <g-progress-circular rotate="-90"
                                 width="1.5"
                                 size="36"
                                 color="#E57373"
                                 value="100"
            >
            </g-progress-circular>
            <div class="progress-remaining">
              300
            </div>
          </div>
        </g-card-title-stub>
        <g-card-text-stub>
          <div>
            <div class="row-flex align-items-start">
              <div style="flex: 0 0 25px; font-weight: 700; font-size: 12px;">
                1x
              </div>
              <div class="flex-equal fs-small-2 pl-1"
                   style="word-break: break-all;"
              >
                1. Pepe 1
                <span class="i text-grey">
                  (Choice 1 - Nhom 1, Choice 1 - Nhom 2)
                </span>
              </div>
              <div class="fs-small-2 ta-right">
                €3.50
              </div>
            </div>
          </div>
          <div>
          </div>
        </g-card-text-stub>
        <g-card-actions-stub>
          <g-btn-bs-stub height="54"
                         width="60"
                         border-color="#C4C4C4"
                         text-color="black"
          >
            <g-icon-stub size="14">
              icon-cross-red
            </g-icon-stub>
          </g-btn-bs-stub>
          <g-btn-bs-stub height="54"
                         background-color="#E0E0E0"
                         class="pending-orders--btn-price"
                         text-color="black"
                         style="flex: 1;"
          >
            <span class="mr-2">
              cash
            </span>
            <span>
              'common.currency' , undefined3.50
            </span>
          </g-btn-bs-stub>
        </g-card-actions-stub>
      </g-card-stub>
      <g-card-stub elevation="0">
        <g-card-title-stub class="pending-orders--title">
          <div class="row-flex align-items-center flex-grow-1">
            <g-icon-stub>
              icon-take-away
            </g-icon-stub>
            <div class="fs-small-2 ml-1"
                 style="max-width: calc(100% - 24px); line-height: 1.2;"
            >
              <span class="fs-small fw-700 text-indigo-accent-2">
                #
              </span>
              dfd -123
            </div>
          </div>
          <div class="row-flex justify-end align-items-center r"
               style="flex: 0 0 auto;"
          >
            <span class="fw-700 fs-small ml-2 mr-2"
                  style="text-transform: uppercase;"
            >
              asap
            </span>
            <g-progress-circular rotate="-90"
                                 width="1.5"
                                 size="36"
                                 color="#E57373"
                                 value="50"
            >
            </g-progress-circular>
            <div class="progress-remaining">
              200
            </div>
          </div>
        </g-card-title-stub>
        <g-card-text-stub>
          <div>
            <div class="row-flex align-items-start">
              <div style="flex: 0 0 25px; font-weight: 700; font-size: 12px;">
                1x
              </div>
              <div class="flex-equal fs-small-2 pl-1"
                   style="word-break: break-all;"
              >
                3. mon test
                <span class="i text-grey">
                </span>
              </div>
              <div class="fs-small-2 ta-right">
                €0.10
              </div>
            </div>
          </div>
          <div>
          </div>
        </g-card-text-stub>
        <g-card-actions-stub>
          <g-btn-bs-stub height="54"
                         width="60"
                         border-color="#C4C4C4"
                         text-color="black"
          >
            <g-icon-stub size="14">
              icon-cross-red
            </g-icon-stub>
          </g-btn-bs-stub>
          <g-btn-bs-stub height="54"
                         background-color="#E0E0E0"
                         class="pending-orders--btn-price"
                         text-color="black"
                         style="flex: 1;"
          >
            <span class="mr-2">
              cash
            </span>
            <span>
              'common.currency' , undefined0.10
            </span>
          </g-btn-bs-stub>
        </g-card-actions-stub>
      </g-card-stub>
    </div>
  </div>
  <div class="kitchen-orders pl-2">
    <div class="header">
      onlineOrder.sentToKitchen
      <g-badge-stub class="ml-1"
                    inline="true"
                    modelvalue="true"
                    color="#F9A825"
      >
      </g-badge-stub>
    </div>
    <div class="content">
      <g-card-stub elevation="0">
        <g-card-title-stub>
          <div class="fs-small-2 ml-1"
               style="max-width: calc(100% - 96px); line-height: 1.2;"
          >
            <span class="fs-small fw-700 text-indigo-accent-2">
              #2
            </span>
            xa -01234
          </div>
          <g-spacer-stub>
          </g-spacer-stub>
          <div class="kitchen-orders__timer">
            <g-icon-stub>
              icon-take-away
            </g-icon-stub>
            <span class="fw-700 fs-small ml-2">
              17:27
            </span>
          </div>
        </g-card-title-stub>
        <g-card-text-stub>
          <div class="row-flex">
            <div class="col-1"
                 style="line-height: 20px;"
            >
              <g-icon-stub color="#9E9E9E"
                           size="20"
              >
                icon-food
              </g-icon-stub>
            </div>
            <div style="padding-top: 2px;">
              <span>
                <span class="fw-700">
                  1x
                </span>
                <span class="mr-3"
                      style="word-break: break-all;"
                >
                  3. mon test
                  <span class="i text-grey fs-small-2">
                  </span>
                </span>
              </span>
            </div>
          </div>
        </g-card-text-stub>
      </g-card-stub>
      <g-card-stub elevation="0">
        <g-card-title-stub>
          <div class="fs-small-2 ml-1"
               style="max-width: calc(100% - 96px); line-height: 1.2;"
          >
            <span class="fs-small fw-700 text-indigo-accent-2">
              #3
            </span>
            xa -0123
          </div>
          <g-spacer-stub>
          </g-spacer-stub>
          <div class="kitchen-orders__timer">
            <g-icon-stub>
              icon-take-away
            </g-icon-stub>
            <span class="fw-700 fs-small ml-2">
              17:27
            </span>
          </div>
        </g-card-title-stub>
        <g-card-text-stub>
          <div class="row-flex">
            <div class="col-1"
                 style="line-height: 20px;"
            >
              <g-icon-stub color="#9E9E9E"
                           size="20"
              >
                icon-food
              </g-icon-stub>
            </div>
            <div style="padding-top: 2px;">
              <span>
                <span class="fw-700">
                  1x
                </span>
                <span class="mr-3"
                      style="word-break: break-all;"
                >
                  4. VIT THAI XOAI | MANGO IN THE „DUCK"
                  <span class="i text-grey fs-small-2">
                  </span>
                </span>
              </span>
            </div>
          </div>
        </g-card-text-stub>
      </g-card-stub>
    </div>
  </div>
  <dialog-complete-order-stub>
  </dialog-complete-order-stub>
  <dialog-text-filter-stub label="Reason">
  </dialog-text-filter-stub>
</div>
`;
